### PR TITLE
ts: add category types

### PR DESCRIPTION
**Pull request by sorja**
Part of #3

### CODE REVIEW

File: src/category/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  index: number
+  props: string
+}
+
+export interface CategoryProps {
+  name: string
+}
+
+export interface CategoryLevels {
+  [key: number]: Category
```
Comment (ramirobg94) [inline src/category/index.ts:13]: Category vs CategoryLevel

**Review by ramirobg94 — CHANGES_REQUESTED** [review]

File: src/category/index.ts (hunk 1 above)
Comment (sorja) [reply src/category/index.ts:13]: I wonder if we need CategoryLevel at all? Isnt it a list of categories?

File: src/category/index.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export interface CategoryLevels {
+  [key: number]: CategoryLevel
+}
+
+export interface CategoryValidation {
+  valid: boolean
+  errors: Array<any>
+  fields: { key: any }
+  warnings: Array<any>
+}
```
Comment (sorja) [inline src/category/index.ts:21]: Should be moved to src/validation

File: src/category/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export interface Category {
+  id: number
+  uuid: string
+  props: CategoryProps
+  published: boolean
+  levels: CategoryLevels
+  validation: CategoryValidation
```
Comment (SteRiccio) [inline src/category/index.ts]: please use the Validation interface here

File: src/category/index.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export interface CategoryProps {
+  name: string
+}
+
+export interface CategoryLevels {
+  [key: number]: Category
+}
+
+export interface CategoryValidation {
```
Comment (SteRiccio) [inline src/category/index.ts]: you don't need CategoryValidation, please use Validation instead

File: src/category/index.ts (hunk 1 above)
Comment (SteRiccio) [reply src/category/index.ts]: Category and CategoryLevel are 2 different things, in this case you have to use CategoryLevel

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: src/category/index.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Validation } from "src/validation";
+
+export interface CategoryLevel {
+  id: number
+  uuid: string
+  index: number
+  props: string
```
Comment (minotogna) [inline src/category/index.ts:7]: this is not a string

**Review by minotogna — CHANGES_REQUESTED** [review]
CategoryItem must be added as well

File: src/category/index.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+export interface CategoryLevelProp {
+  name: string
+}
+
+export interface CategoryLevel {
+  id: number
+  uuid: string
+  index: number
+  props: CategoryLevelProp
```
Comment (minotogna) [inline src/category/index.ts:17]: this should be `CategoryLevelProps`

File: src/category/index.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+export interface CategoryPropsItemExtraDef {
```
Comment (minotogna) [inline src/category/index.ts:20]: call this CategoryItemExtraDef.
add the missing properties

File: src/category/index.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  dataType: CategoryPropsItemExtraDefDataType
```
Comment (minotogna) [inline src/category/index.ts:21]: for simple properties you can use `‘text’|'number’|'geometryPoint’`

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by SteRiccio — APPROVED** [review]

File: src/index.ts [hunk 9]
```diff
@@ -9,3 +17,6 @@ export type { Validation } from './validation'
 
 // Enums
 export { UserStatus, UserTitle } from './auth/user'
+
+// Generics
+export { ArenaObject } from './common'
```
Comment (minotogna) [inline src/index.ts:22]: export type

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by ramirobg94 — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**